components/electric/lib/electric/ddlx/parse/assign_parser.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Electric.DDLX.Parse.AssignParser do
+  alias Electric.DDLX.Parse.Element
+  alias Electric.DDLX.Command.Assign
+  import Electric.DDLX.Parse.Common
+
+  @keywords ["electric", "assign", "to", "if"]
+
+  @elements [
+    %Element{required: true, type: "keyword", options: ["electric"], name: nil},
+    %Element{
+      required: true,
+      type: "kv",
+      options: ["assign"],
+      name: "role",
+      valueType: [:string, :collection, :name]
+    },
+    %Element{required: true, type: "kv", options: ["to"], name: "user", valueType: :name},
+    %Element{required: false, type: "kv", options: ["if"], name: "if", valueType: :collection}
+  ]
+
+  use Electric.DDLX.Parse.Common
+
+  def matches(statement) do
+    String.starts_with?(statement, "electric assign")
+  end
```

Comment:
It's probably more a matter of style than performance, but I'd use pattern matching here as a more efficient and idiomatic construct.
```suggestion
  def matches(statement) do
    match?("electric assign" <> _, statement)
  end
```